packages/models/src/delegation/delegationEvents.ts

```diff
@@ -0,0 +1,60 @@
+import { z } from "zod";
+import { match } from "ts-pattern";
+
+import {
+  DelegationSubmittedV2,
+  DelegationApprovedV2,
+  DelegationRejectedV2,
+  DelegationRevokedV2,
+} from "../gen/v2/delegation/events.js";
+import { protobufDecoder } from "../protobuf/protobuf.js";
+import { EventEnvelope } from "../events/events.js";
+
+export const DelegationEventV2 = z.discriminatedUnion("type", [
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationSubmittedV2"),
+    data: protobufDecoder(DelegationSubmittedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationApprovedV2"),
+    data: protobufDecoder(DelegationApprovedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationRejectedV2"),
+    data: protobufDecoder(DelegationRejectedV2),
+  }),
+  z.object({
+    event_version: z.literal(2),
+    type: z.literal("DelegationRevokedV2"),
+    data: protobufDecoder(DelegationRevokedV2),
+  }),
+]);
+
+export type DelegationEventV2 = z.infer<typeof DelegationEventV2>;
```

Comment:
in this case we only have a V2 version, so V1 is not implemented and a generic `DelegationEvent` type is not required.

Reply:
I would stick to the convention of having an `-Event` type which is a discriminated union of all the event versions, in this case only one (`DelegationEventV2`). 
This way, adding another version in the future will be much easier